src/interceptors/ClientRequest/MockHttpSocket.ts

```diff
@@ -298,7 +306,7 @@ export class MockHttpSocket extends MockSocket {
   }
 
   private mockConnect(): void {
-    this.emit('lookup', null, '::1', 6, this.connectionOptions.host)
+    this.emit('lookup', null, '127.0.0.1', 4, this.connectionOptions.host)
```

Comment:
I think setting `this.address` should be done in `mockConnect()` for mocked responses. This will couple the `.address()` function withe actual address we emit in the lookup even so those two stay consistent (right now, they are not).

```
const address = {
  host: '...',
  family: '...',
  port: '...'
}
this.address = () => address

this.emit('lookup', null, address.host, address.family, this.connectionOptions.host)
```

Reply:
Wonderful! I'm ashamed that I didn't think about it myself.😅

Reply:
You think of a ton of things that never crossed my mind 🙌 The power of collaboration.